Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowsFolderTreeItem.cs

```diff
@@ -138,14 +139,19 @@ List<ITreeViewItem> ITreeViewItem.Childrens()
             else
             {                
                 if (WorkSpace.Instance.BetaFeatures.BFUseSolutionRepositry)
-                {                    
+                {
+                    AddsubFolders(mRepositoryFolder, Childrens);
+                    if (mViewMode == eBusinessFlowsTreeViewMode.FoldersOnly)
+                        return Childrens;                        
                     BFs = mRepositoryFolder.GetFolderItems();
-                    AddsubFolders(mRepositoryFolder, Childrens);                    
+                                       
                 }
                 else
-                {                    
-                    BFs = App.LocalRepository.GetSolutionBusinessFlows(specificFolderPath: Path);
+                {   
                     AddsubFolders(Path, Childrens);
+                    if (mViewMode == eBusinessFlowsTreeViewMode.FoldersOnly)
+                        return Childrens;
+                    BFs = App.LocalRepository.GetSolutionBusinessFlows(specificFolderPath: Path);
                 }                                
```

Comment:
i don't think this the solution for Folder selection because this is specific handling only for BF's tree, what if we will want to select Run Set folder? we don't want to modify all Folders tree items.
solution should be on TreeSelectionPage which will get param to show only folders, this config will pass to UCTree which will avoid adding treeitems which are not Folders (you can use the isExpandable flag to know which is Folder) in generic way.

Reply:
i could have done that, but the issue is in case of heavy solutions,get all children will get all the child elements and then only we will need to filter the ones which are folder items, which i snot good solution performance wise.

Reply:
This option also thought, but the problem is ITreeViewItem.Childrens() is called when the tree is expanded and since ITreeViewItem is interface I cannot define enum/bool at this level. So had to add it in specific treeitem.